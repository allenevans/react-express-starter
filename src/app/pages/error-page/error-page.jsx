import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import './error-page.scss';
import Layout from '../layout';
import { PageContext } from '../../contexts/page-context/page-context';

const errors = {
  404: {
    title: 'Page not found',
    desc: 'This page is either no longer available or or the URL has been entered incorrectly.',
  },
  500: {
    title: 'Server error',
    desc: 'There was a problem processing your request.',
  },
};

export const ErrorPage = ({ server }) => {
  const { error } = server;
  const responseStatus = error ? 500 : 404;
  const mappedError = errors[responseStatus];
  const errorMessage = error ? error.message : mappedError.title;
  const errorStack = error ? error.stack : null;

  return (
    <Layout className="error-page page">
      <PageContext.Consumer>{({ head, statusCode }) => {
        head(<title>{mappedError.title}</title>);
        statusCode(responseStatus);
      }}</PageContext.Consumer>

      <h1>{errorMessage}</h1>
      <div>{errorStack}</div>
    </Layout>
  );
};

ErrorPage.defaultProps = {
  server: {},
};

ErrorPage.propTypes = {
  server: PropTypes.object,
};

/* istanbul ignore next */
const mapStateToProps = ({ server }) => ({
  server,
});


export default connect(mapStateToProps)(ErrorPage);
