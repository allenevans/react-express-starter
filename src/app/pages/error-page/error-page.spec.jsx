import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { ErrorPage } from './error-page';

const renderer = new ReactShallowRenderer();

describe('error-page component', () => {
  it('should render the component', () => {
    const tree = renderer.render(<ErrorPage />);

    expect(tree).toMatchSnapshot();
  });
});
