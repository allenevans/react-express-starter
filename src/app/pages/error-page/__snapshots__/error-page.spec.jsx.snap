// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error-page component should render the component 1`] = `
<Connect(Layout)
  className="error-page page"
>
  <React.Fragment>
    <Context.Consumer>
      [Function]
    </Context.Consumer>
    <h1>
      Page not found
    </h1>
    <div>
      This page is either no longer available or or the URL has been entered incorrectly.
    </div>
  </React.Fragment>
</Connect(Layout)>
`;
